harness/determined/deploy/aws/gen_vcpu_mapping.py

```diff
@@ -0,0 +1,47 @@
+import argparse
+import json
+from pathlib import Path
+from typing import Dict, Iterable, List, Tuple
+
+import boto3
+import yaml
+
+
+def _fetch_vcpu_mapping() -> Iterable[Tuple[str, Dict]]:
+    # Price List api is only available in us-east-1 and ap-southeast-1.
+    client = boto3.client("pricing", region_name="us-east-1")
+    for page in client.get_paginator("get_products").paginate(ServiceCode="AmazonEC2"):
+        for sku_str in page["PriceList"]:
+            sku_data = json.loads(sku_str)
+            try:
+                attributes = sku_data["product"]["attributes"]
+                data = {
+                    "instanceType": attributes["instanceType"],
+                    "vcpu": int(attributes["vcpu"]),
+                }
+
+                if "gpu" in attributes:
+                    data["gpu"] = int(attributes["gpu"])
+
+                if "instanceType" in data:
+                    yield (attributes["instanceType"], data)
+            except KeyError:
+                pass
+
+
+def fetch_vcpu_mapping() -> List[Dict]:
+    return [v for (_, v) in sorted(dict(_fetch_vcpu_mapping()).items())]
```

Comment:
Oh, I just remembered that the YAML library sorts keys by default, so this could simply be `dict(_fetch_vcpu_mapping())` if it were stored as a dict. But it works fine as it is.

Reply:
interesting, looks like it does sort them indeed. but I am not going to put too much trust into yaml serializer.

Reply:
Hah, that's fair.